pandas/tests/frame/indexing/test_where.py

```diff
@@ -1017,3 +1020,15 @@ def test_where_producing_ea_cond_for_np_dtype():
         {"a": Series([pd.NA, pd.NA, 2], dtype="Int64"), "b": [np.nan, 2, 3]}
     )
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "replacement", [0.001, True, "snake", DATETIME_JAN_1_1900_OPTIONAL_TZ]
+)
+def test_where_int_overflow(replacement):
+    # GH 31687
+    df = DataFrame([[1.0, 2e19, "nine"], [np.nan, 0.1, None]])
```

Comment:
ideally for regression tests it's better to keep them as close as possible to the original - any reason to use `2e19` instead of `2e25`?

Reply:
I think I reduced it down to the smallest order of magnitude that still had the problem in pandas 1.4.1, but I don't have strong feelings about using that instead of the 2e25 from the original.